fe/fe-core/src/main/java/org/apache/doris/rewrite/ExprRewriter.java

```diff
@@ -55,6 +58,12 @@ public Expr rewrite(Expr expr, Analyzer analyzer) throws AnalysisException {
                 rewrittenExpr = applyRuleRepeatedly(rewrittenExpr, rule, analyzer);
             }
         } while (oldNumChanges != numChanges_);
+        if (expr instanceof BinaryPredicate) {
+            Expr valueExpr = ((BinaryPredicate) expr).getBinding();
+            if(valueExpr != null && valueExpr.getType() == Type.DATETIME && valueExpr instanceof CastExpr) {
+                throw new AnalysisException("invalid date type :" + valueExpr.toSql());
```

Comment:
I think throwing a exception is not the best solution of this problem, may be rewrite the expr `day = '2020-10-32'` to something like `1 = 0` in where statement is better. We should discuss it first.

Reply:
> I think throwing a exception is not the best solution of this problem, may be rewrite the binary expr `day = '2020-10-32'` to something like contant expr BoolLiteral[false] in where statement is better. We should discuss it first.

ok，I will modify it;